### PR TITLE
Add linting for insensitive and inconsiderate language

### PR DESCRIPTION
**Pull request by rickhanlonii**
## Overview

This PR number [1337](https://en.wikipedia.org/wiki/Leet), adds [alexjs](https://alexjs.com/) linting to the `/docs` folder to prevent insensitive and inconsiderate language from landing on master.

- Adds Alex
- Configures Alex rules
- Adds to circleci tests as land-blocking

## Blocker
This [PR](https://github.com/rickhanlonii/react-native-website/pull/1) fixes 193 lint issues that exist in the current docs. Once that PR is merged into this one, we can merge this PR into master.

PRs are separated to make reviewing this one easier.

## What about versioned docs
Right now we're only linting the master docs in the `/docs` folder. Older docs in the `website/versioned_docs` directory are not linted.

Once we fix those docs, we can enable linting everywhere.

### CODE REVIEW

**Comment by rickhanlonii** [comment]
Merge blocked on https://github.com/rickhanlonii/react-native-website/pull/1, but feel free to review and approve this one first

File: .alexrc.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * LICENSE file in the root directory of this source tree.
+ */
+
+exports.allow = [
+  // We frequently refer to form props by their name "disabled".
+  // Ideally we would alex-ignore only the valid uses (PRs accepted).
+  "invalid",
+
+  // Unfortunately "watchman" is a library name that we depend on.
+  "watchman-watchwoman"
```
Comment (rickhanlonii) [inline .alexrc.js]: There are just two rules that we're opting out of

File: .alexrc.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+];
+
+// Use a "maybe" level of profanity instead of the default "unlikely".
+exports.profanitySureness = 1;
```
Comment (rickhanlonii) [inline .alexrc.js]: When this is set to 0 it is too eager, it catches words like "color", "execute", and "fire" which in our context are super unlikely to be profane and highly likely to be valid

**Comment by rickhanlonii** [comment]
Shout out @wooorm for the help and alexjs ✨

File: .alexrc.js (hunk 2 above)
Comment (TheSavior) [reply .alexrc.js]: Should you put your reasoning as a comment in the code?

**Comment by TheSavior** [comment]
Wow, lots of warnings on the lint job! I'm glad we are cleaning this up. Thanks for setting this up. 

It would be great if you could also hook this up to also comment back on the PR at the line. We have a bot set up to do that on the main RN repo for eslint issues.

**Comment by rickhanlonii** [comment]
Update: I merged in the changes from https://github.com/rickhanlonii/react-native-website/pull/1 and added a new `yarn lintv` command to help with linting the versioned docs

**Comment by pvinis** [comment]
I love the removal of "just" and "simply", but I don't think "easy" -> "quick" or "simple" -> "basic" is an improvement 😬

**Comment by rickhanlonii** [comment]
@pvinis I hear you.

The way I think about it, "simple" seems to be a characteristic of knowledge while "basic" is a characteristic of form. Something can be "basic" (is made of few parts) but not "simple" (easily understood). By focusing on the form instead of the knowledge characteristic then the docs are more inclusive

If you feel strongly I would recommend reaching out to the retext repo and getting it added there and we would address it in the repo here ✨

**Pull request merged**